queries/earthfile/highlights.scm

```diff
@@ -0,0 +1,126 @@
+(string_array
+  "," @punctuation.delimiter)
+
+(string_array
+  [
+    "["
+    "]"
+  ] @punctuation.bracket)
+
+[
+  "ARG"
+  "AS LOCAL"
+  "BUILD"
+  "CACHE"
+  "CMD"
+  "COPY"
+  "DO"
+  "ENTRYPOINT"
+  "ENV"
+  "EXPOSE"
+  "FROM DOCKERFILE"
+  "FROM"
+  "FUNCTION"
+  "GIT CLONE"
+  "HOST"
+  "IMPORT"
+  "LABEL"
+  "LET"
+  "PROJECT"
+  "RUN"
+  "SAVE ARTIFACT"
+  "SAVE IMAGE"
+  "SET"
+  "USER"
+  "VERSION"
+  "VOLUME"
+  "WORKDIR"
+] @keyword
+
+(for_command
+  [
+    "FOR"
+    "IN"
+    "END"
+  ] @keyword.repeat)
+
+(if_command
+  [
+    "IF"
+    "END"
+  ] @keyword.conditional)
+
+(elif_block
+  "ELSE IF" @keyword.conditional)
+
+(else_block
+  "ELSE" @keyword.conditional)
+
+(import_command
+  [
+    "IMPORT"
+    "AS"
+  ] @keyword.import)
+
+(try_command
+  [
+    "TRY"
+    "FINALLY"
+    "END"
+  ] @keyword.exception)
+
+(wait_command
+  [
+    "WAIT"
+    "END"
+  ] @keyword)
+
+(with_docker_command
+  [
+    "WITH DOCKER"
+    "END"
+  ] @keyword)
+
+[
+  (comment)
+  (line_continuation_comment)
+] @comment @spell
+
+[
+  (target_ref)
+  (target_artifact)
+  (function_ref)
+] @function
+
+(target
+  (identifier) @function)
+
+[
+  (double_quoted_string)
+  (single_quoted_string)
+] @string
+
+(unquoted_string) @string.special
+
+(escape_sequence) @string.escape
+
+(variable) @variable
+
+(expansion
+  [
+    "$"
+    "{"
+    "}"
+    "("
+    ")"
+  ] @punctuation.special)
+
+(build_arg
+  (variable) @variable.parameter) @variable.parameter
```

Comment:
```suggestion
  (variable) @variable.parameter)
```

Reply:
in `--foo=bar` that would make `--` and `foo` be highlighted differently:

<img width="88" alt="Capture d’écran 2024-04-04 à 17 24 19" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/325401/56834bf1-cb09-4326-992e-b0370b1fe627">

I clearly prefer to have it highlighted uniformly, but I’ll change it if that’s a common practice — just checking that we are well aligned :)

Reply:
Why not 
```
(build_arg) @variable.parameter
```
then?

Reply:
As long as it doesn't capture the value.

Reply:
(or `(buildarg (variable)) @variable.parameter`, if it makes a difference)

Reply:
neither `(build_arg) @variable.parameter` or `(buildarg (variable)) @variable.parameter` produce an uniformly highlighted `--foo`. That’s because the `build_arg` node contains a `variable` node which is already highlighted as `@variable`.

`(build_arg (variable) @variable.parameter) @variable.parameter` applies the `@variable.parameter` to both the `build_arg` node, with is visible as the `--` and the `variable` node which is visible as `foo`.

<img width="88" alt="Capture d’écran 2024-04-04 à 22 23 09" src="https://github.com/nvim-treesitter/nvim-treesitter/assets/325401/37a362c7-75c0-42a1-bb04-dfbaeab1579d">

The value is a `string` which itself is made of one or more `unquoted_string`, `single_quoted_string` or `double_quoted_string` which are themselves highlighted with other queries.

Is there something wrong with that query?

Reply:
So what im seeing is there is no dedicated anon node for - and --

Reply:
there is a anonymous `--` node. Is it better to highlight that node instead of the whole `build_arg` node?

something like:

```
(build_arg
  [
    "--"
    (variable)
  ] @variable.parameter)
```

Reply:
Yes

Reply:
ok, works for me 👍

Reply:
Is there a documentation with the good practices on writing tree-sitter queries that I may have missed?

Reply:
CONTRIBUTING.md is the best documentation we currently have.